### PR TITLE
UltraVNC Repeater

### PR DESCRIPTION
**Pull request by JimmyS83**
Maintainer: Jimmy / @JimmyS83 / robutek@gmail.com
Compile tested: OpenWRT 15.05.1
Run tested: OpenWRT 15.05.1 / TP-LINK WR741ND

Description:
Port of UltraVNC Repeater for linux by Jari Korhonen. Based on latest 0.17 version.

### CODE REVIEW

**Comment by diizzyy** [comment]
* Where is 0.15 and later? http://jtko.mbnet.fi/uvncrepeater/
* No PKG_LICENSE entry in Makefile
* Unless it's ~one-two file(s) to compile source code should be pulled from a repo instead of being hosted within the package repo
* What does it offer compared to a regular TCP wrapper/proxy?

**Comment by JimmyS83** [comment]
Hi Dizzy,

_Where is 0.15 and later? http://jtko.mbnet.fi/uvncrepeater/_
I was searching for the latest version avaible. And I found http://www.wisdomsoftware.gr/download/uvncrep017-ws.tar.gz (based on http://forum.ultravnc.info/viewtopic.php?t=26766 article).

_No PKG_LICENSE entry in Makefile_
I will add GNU GPL

_Unless it's ~one-two file(s) to compile source code should be pulled from a repo instead of being 
hosted within the package repo_
Unfortunatelly I cant provide stable and long lasting repository. Is that really needed? In other case, I can make patch based on existing hosting for linux source, but this package will be dependent on that (I know that repo works that way, but for me it was just a few kB)

_What does it offer compared to a regular TCP wrapper/proxy?_
I am not aware of wrapper/proxy. I am using uVNC for years, and repeater helps me a big time connecting to NATet/Secured environments. I usually work through 23/80 ports, and it works flawlesly with UltraVNC clients.

_You need to squash you commits_
I try to figure out, what does it means and how to do that

_Better choice? https://github.com/XSoftBG/repeater_
Is it already included in OpenWRT? Works with 100% UltraVNC client? Before I flashed my router, I just using Windows version of uVNC distributer/repeater (native). When I flash to OpenWRT, I just compile this linux version for myself, and using it all day almost. I just want to contribute OpenWRT and make this possible for other users, to give them possibility to use UltraVNC stuff out of the box through opkg...

**Comment by JimmyS83** [comment]
**Fixed** _Where is 0.15 and later? http://jtko.mbnet.fi/uvncrepeater/_ 
**Fixed** _No PKG_LICENSE entry in Makefile_
**Fixed** _Unless it's ~one-two file(s) to compile source code should be pulled from a repo instead of being hosted within the package repo_

_You need to squash you commits_
I tried to fight with git a long time today, but I cant manage that. Hope it isnt such a big deal, now is package prepared by your demands..

**Comment by jp-bennett** [comment]
https://ariejan.net/2011/07/05/git-squash-your-latests-commits-into-one/
It's really quite preferable to squash commits, and not too difficult once you figure it out for the first time.  I remember struggling with the concept for a couple hours at first, too.

**Comment by hnyman** [comment]
It is actually a big issue...  This will not be merged in in the current form.

Now you use 10 commits to create 3 files. there should be just 1 commit.
And there is no proper maintainer in the Makefile.
All commits need to have the signed-off-by line.

You should squash the commits together. It is not that hard with "git rebase -i". Advice in https://github.com/openwrt/packages/blob/master/CONTRIBUTING.md#advice-on-pull-requests
I tried to explain that earlier today in https://github.com/openwrt/packages/pull/3528#issuecomment-261703071

**Comment by JimmyS83** [comment]
Allright, I was so close yesterday, but I didnt know about -f by push command. Thanks hnyman.

Anyway, now its 1 commit, properly signed off, based on Jaris 0.14, using Jaris repository online, just 3 files. What do you say now? :)

File: net/vncrepeater/Makefile [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+# See /LICENSE for more information.
+#
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=vncrepeater
+PKG_VERSION:=0.14
+PKG_RELEASE:=1
+PKG_MAINTAINER:=JimmyS83 
+PKG_LICENSE:=GNU GPL
```
Comment (hnyman) [inline net/vncrepeater/Makefile]: Two minor comments: 

* You should check the license code from the SPDX list and use the corerct GNU GPL abbreviation depending on the upstream license, e.g. GPL-2.0
See  https://github.com/openwrt/packages/blob/master/CONTRIBUTING.md#common-license-tags-short-list

File: net/vncrepeater/Makefile [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+define Build/Compile
+	$(call Build/Compile/Default, \
+		CC="$(TARGET_CXX)" \
+		CFLAGS="$(TARGET_CFLAGS) $(TARGET_CPPFLAGS) -fno-rtti" \
+		repeater \
+	)
+endef
+
+define Package/vncrepeater/install	
+	$(INSTALL_DIR) $(1)/usr/sbin $(1)/etc
```
Comment (hnyman) [inline net/vncrepeater/Makefile]: * Directory  ``` $(1)/etc ``` is created twice. On this line and then also two lines later. It is probably more clear if you remove it from this first line.

File: net/vncrepeater/Makefile [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=vncrepeater
+PKG_VERSION:=0.14
+PKG_RELEASE:=1
+PKG_MAINTAINER:=JimmyS83
```
Comment (hnyman) [inline net/vncrepeater/Makefile]: And also the maintainer field needs your normal full name and email address

**Comment by JimmyS83** [comment]
Hello,
Just fixed all three things :)

**Pull request merged**